internal/proxy/oauthproxy.go

```diff
@@ -375,6 +375,27 @@ func (p *OAuthProxy) IsWhitelistedRequest(req *http.Request) bool {
 	return false
 }
 
+// runValidatorsWithGracePeriod runs all validators and upon finding errors, checks to see if the
+// auth provider is explicity denying authentication or if it's merely unavailable. If it's unavailable,
+// we check whether the session is within the grace period or not to determine the specific error we return.
+func (p *OAuthProxy) runValidatorsWithGracePeriod(session *sessions.SessionState) (err error) {
```

Comment:
It seems like we only every run validators with the grace period in the proxy. It also seems strange to me that the function that is responsible for running the validators hangs of the options struct, which doesn't feel like it has much to do with running the validators?

Can we move the running the validators to the OAuthProxy struct and have it just always run the grace period? I think this simplification will better delineate the responsibility of these various structs and simplifying the confusing naming.

Reply:
On the point of `options` in particular, I think in this case that is the 'options' _package_, the naming of which hung around from previous times. The reason that we're using the package there specifically is to call the `RunValidators` helper function, with the validators that are stored in the OAuthProxy struct being passed in.


I think now that these are more defined 'validators' the package could surely do with a name update.

Reply:
ah, right -- I think the point still holds regardless of the package vs struct.

Reply:
I've renamed the 'options' package to 'validators' to better represent its responsibility, though given that change I'm not sure if we'd still like to change the how we run the validators themselves?

Right now we're running these validators in two slightly different ways within the proxy:
- by calling the `validators.RunValidators` function in the `OAuthCallback` method (https://github.com/buzzfeed/sso/pull/275/files#diff-0f5a3bf10e8922fb438df82f694747a5L572)
- by calling the `(oauthproxy.)runValidatorsWithGracePeriod` method in the `Authenticate` method, which calls the `validators.RunValidators` function before adding some extra logic to handle provider grace periods
(https://github.com/buzzfeed/sso/pull/275/files#diff-0f5a3bf10e8922fb438df82f694747a5R781)